tests/suites/host_test.function

```diff
@@ -432,6 +432,43 @@ static void write_outcome_result(FILE *outcome_file,
     fflush(outcome_file);
 }
 
+#if defined(__unix__) ||                                \
+    (defined(__APPLE__) && defined(__MACH__))
+#define MBEDTLS_HAVE_CHDIR
+#endif
+
+#if defined(MBEDTLS_HAVE_CHDIR)
```

Comment:
If this is going to be a best-effort function, why not unconditionally define it, and just have it do nothing (apart from maybe print a warning) on platforms like Windows?

Reply:
What do you mean by “it”? The `try_chdir` function? I guess that could make sense, but then it wouldn't be best-effort anymore, it would be best-effort-if-supported. I guess that's not a big deal either way since I don't expect the function to have more than one call site.

Reply:
Yes, `try_chdir`. IMO it's probably slightly simpler to lose the '#if defined ... ' at the call site, but agree not a big deal - not worth iterating for this.

Reply:
I'm not fully convinced, but I realized this needed rework anyway, so I made this change.